### PR TITLE
Support latest build packages, migrate to null safety

### PR DESCRIPTION
**Pull request by simolus3**
Support the latest version of the build system and migrate to the implementation to null safety.

### CODE REVIEW

**Comment by simolus3** [comment]
Sorry if I'm bothering you here @nshahan @luisvt, but could you maybe take a look at this? I think it would be great if we could use this package with the latest version of the build system.

File: pubspec.yaml [hunk 1]
```diff
@@ -10,14 +10,14 @@ homepage: https://github.com/dart-league/sass_builder
...
-  sdk: '>=2.0.0-dev.51 <3.0.0'
+  sdk: '>=2.12.0 <3.0.0'
 
 dependencies:
-  build: '>=0.12.5 <2.0.0'
-  build_config: '>=0.3.0 <0.5.0'
-  path: ^1.4.1
-  sass: '>=1.11.0 <2.0.0'
+  build: ^2.0.0
+  build_config: '>=0.3.0 <2.0.0'
```
Comment (jakemac53) [inline pubspec.yaml]: We should probably just bump this to `^1.0.0` as well? Best to just force all deps onto null safety even if they aren't directly imported today.

**Review by jakemac53 — APPROVED** [review]
This LGTM from a package:build/nnbd perspective with the one small nit.

@nshahan can you merge/publish?

**Review by nshahan — APPROVED** [review]
LGTM

**Comment by nshahan** [comment]
@simolus3 Thanks for the migration. Sorry about the delayed response but I just published a new version with your changes.

**Pull request merged**